packages/kas/src/nodes.ts

```diff
@@ -874,22 +904,24 @@ export class Mul extends Seq {
             var isRational =
                 numbers[i] instanceof Rational && !(numbers[i] instanceof Int);
             if (isRational && others.length > 0 && inverses.length > 0) {
-                var withThisRemoved = numbers.slice();
+                var withThisRemoved: Expr[] = numbers.slice();
                 withThisRemoved.splice(i, 1);
                 var newTerms = withThisRemoved.concat(inverses).concat(others);
                 return numbers[i].tex() + new Mul(newTerms).tex();
             }
         }
 
         numbers = _.compact(
-            _.map(numbers, function (term) {
-                var hasDenom =
-                    term instanceof Rational && !(term instanceof Int);
+            _.map(numbers, (term) => {
                 var shouldPushDown =
                     !term.hints.fraction || inverses.length > 0;
-                if (hasDenom && shouldPushDown) {
+                if (
+                    term instanceof Rational &&
+                    !(term instanceof Int) &&
```

Comment:
**nit:** I kind of like having `hasDenom` as I think it makes intent clearer. Wouldn't lose sleep either way though.

Reply:
I hear you.  TS unfortunately isn't able to follow the logic when it's expressed that way.